activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -527,7 +529,9 @@ def delete_all(dependent = nil)
       #
       #   Pet.find(1) # => Couldn't find Pet with id=1
       def destroy_all
-        @association.destroy_all
+        count = @association.destroy_all
```

Comment:
`@association.destroy_all` returns records, not count.
How about using `@association.destroy_all.tap { reset_scope }` to not care about what is returning?